File: packages/on_audio_query/on_audio_query_ios/ios/Classes/controllers/QueryController.swift
import Flutter

public class QueryController {
    var call: FlutterMethodCall
    var result: FlutterResult
    
    init(call: FlutterMethodCall, result: @escaping FlutterResult) {
        self.call = call
        self.result = result
    }
    
    //This method will sort call according to request.
    public func chooseMethod() {
        // All necessary method to this plugin support both platforms, only playlists
        // are limited when using [IOS].
        switch call.method {
        case "queryAudios":
            AudiosQuery(call: call, result: result).queryAudios()
        case "queryAlbums":
            AlbumsQuery(call: call, result: result).queryAlbums()
        case "queryArtists":
            ArtistsQuery(call: call, result: result).queryArtists()
        case "queryGenres":
            GenresQuery(call: call, result: result).queryGenres()
        case "queryPlaylists":
            PlaylistsQuery(call: call, result: result).queryPlaylists()
        case "queryArtwork":
            ArtworkQuery(call: call, result: result).queryArtwork()
        // The playlist for [IOS] is completely limited, the developer can only:
        //   * Create playlist
        //   * Add item to playlist (Unsuported, for now)
        //
        // Missing methods:
        //   * Rename playlist
        //   * Remove playlist
        //   * Remove item from playlist
        //   * Move item inside playlist
        case "createPlaylist":
            PlaylistsController(call: call, result: result).createPlaylist()
        case "addToPlaylist":
            PlaylistsController(call: call, result: result).addToPlaylist()
        default:
            // All non suported methods will throw this error.
            result(FlutterMethodNotImplemented)
        }
    }
}

File: packages/on_audio_query/on_audio_query_android/android/src/main/kotlin/com/lucasjosino/on_audio_query/controllers/QueryController.kt
package com.lucasjosino.on_audio_query.controllers

import android.content.Context
import com.lucasjosino.on_audio_query.methods.queries.*
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

class QueryController(
    private val context: Context,
    private val call: MethodCall,
    private val result: MethodChannel.Result
) {

    //
    fun call() {
        when (call.method) {
            //Query methods
            "queryAudios" -> AudiosQuery().init(context, result, call)
            "queryAlbums" -> AlbumsQuery().init(context, result, call)
            "queryArtists" -> ArtistsQuery().init(context, result, call)
            "queryPlaylists" -> PlaylistsQuery().init(context, result, call)
            "queryGenres" -> GenresQuery().init(context, result, call)
            "queryArtwork" -> ArtworkQuery().queryArtwork(context, result, call)
            //Playlists methods
            "createPlaylist" -> PlaylistController().createPlaylist(context, result, call)
            "removePlaylist" -> PlaylistController().removePlaylist(context, result, call)
            "addToPlaylist" -> PlaylistController().addToPlaylist(context, result, call)
            "removeFromPlaylist" -> PlaylistController().removeFromPlaylist(
                context,
                result,
                call
            )
            "renamePlaylist" -> PlaylistController().renamePlaylist(context, result, call)
            "moveItemTo" -> PlaylistController().moveItemTo(context, result, call)
            // Called if the requested method doesn't exist.
            else -> result.notImplemented()
        }
    }
}